src/realm/array_decimal128.cpp

```diff
@@ -65,10 +152,16 @@ void ArrayDecimal128::move(ArrayDecimal128& dst_arr, size_t ndx)
 {
     size_t elements_to_move = m_size - ndx;
     if (elements_to_move) {
+        if (m_width >= dst_arr.m_width) {
+            dst_arr.upgrade_leaf(m_width);
+        }
+        else {
+            upgrade_leaf(dst_arr.m_width);
+        }
         const auto old_dst_size = dst_arr.m_size;
-        dst_arr.alloc(old_dst_size + elements_to_move, sizeof(Decimal128));
-        Decimal128* dst = reinterpret_cast<Decimal128*>(dst_arr.m_data) + old_dst_size;
-        Decimal128* src = reinterpret_cast<Decimal128*>(m_data) + ndx;
+        dst_arr.alloc(old_dst_size + elements_to_move, m_width);
+        auto dst = dst_arr.m_data + old_dst_size * m_width;
+        auto src = m_data + ndx * m_width;
         memmove(dst, src, elements_to_move * sizeof(Decimal128));
```

Comment:
Can you double check this?
```suggestion
        memmove(dst, src, elements_to_move * m_width);
```

Reply:
Good catch